apps/api/src/controllers/languages.controller.test.ts

```
@@ -0,0 +1,565 @@
+      if (selectArg && typeof selectArg === 'object' && 'count' in selectArg) {
+        return mockDbSelectForCount; // Use the count-specific mock
+      }
+      // Default select behavior
+      const self = {
+        from: mock().mockReturnThis(),
+        where: mock().mockReturnThis(),
+        orderBy: mock().mockReturnThis(),
+        limit: mock().mockReturnThis(),
+        offset: mock().mockReturnThis(),
+        // Add then directly to the object returned by select() to simulate promise-like behavior for query execution
+        then: (callback: any) => callback([]), // Default to returning empty array for a select query
```

Biome (1.9.4)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)